Add UI improvments

CajetanRodrigues opened the pull request:
### Dashboard and Login Component Improvements

- Improved the behavior of the side navbar based on appropriate conditions and user login status.
- proof-checked the responsiveness for mobile and desktop screens.
- Standardized fonts across the app.
- Removed redundant code.

![dashboard login](https://user-images.githubusercontent.com/37682760/63351652-4061f280-c37d-11e9-987b-c93c2851d2c0.gif)

ankit01ojha commented on src/app/services/authentication.service.ts:
```diff
@@ -12,25 +12,15 @@ const httpOptions = {
-  key = 'token';
   constructor(private http: HttpClient, private appService: AppService) { }
    login(username: string, password: string) {
      if (username === 'mifos' && password === 'password') {
```
@CajetanRodrigues hardcoded, intergrate APIs.

CajetanRodrigues replied:
Ok I will sort that out

CajetanRodrigues replied:
Hi Ankit, I have incorporated all the changes you have suggested.

ankit01ojha commented on src/app/services/authentication.service.ts:
```diff
@@ -1,36 +1,46 @@
+          return false;
+        }
+      });
+      if (username === 'mifos' && password === 'password') {
```
@CajetanRodrigues hardcoded.

ankit01ojha commented on src/app/services/authentication.service.ts:
```diff
@@ -1,36 +1,46 @@
+          return false;
+        }
+      });
+      if (username === 'mifos' && password === 'password') {
```
@CajetanRodrigues

CajetanRodrigues replied:
I have resolved it

ankit01ojha commented:
@CajetanRodrigues, sometimes I am getting this on the landing page 
![Uie2](https://user-images.githubusercontent.com/21126132/63481656-c54f2800-c4b3-11e9-9c8a-4a61813be64d.png)

CajetanRodrigues commented:
> @CajetanRodrigues, sometimes I am getting this on the landing page
> ![Uie2](https://user-images.githubusercontent.com/21126132/63481656-c54f2800-c4b3-11e9-9c8a-4a61813be64d.png)

I have resolved it

ankit01ojha commented on src/app/app.component.html:
```diff
@@ -35,12 +36,12 @@ <h1 class="example-app-name">MIFOS</h1>
                      <mat-card id="profile-background">
                         <img id="profile-picture" src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxASEhUTExAQFRMTEA8VFxIVDw8VFRcXFRUWFhUXGBUYHSggGBolHRUVITEhJSorLi4uFx8zODMsNygtLi4BCgoKDg0OFxAQGCsdHR0tLi0tLS0tLS0tLS0rLS0rLS0tLi0tLS0tLS0tLS0tLS0tKy0tLS0tLS0tLS0tLS0tLf/AABEIAOEA4QMBEQACEQEDEQH/xAAbAAEAAgMBAQAAAAAAAAAAAAAABQYBBAcDAv/EAEgQAAECAwUEBQYLBgUFAAAAAAEAAgMRMQQSIWFxBQZBURMig ...
-                       <p>  Welcome, Smith R </p>
+                       <p class="font-style" data-aos="fade-up">  Welcome, Smith R </p>
```
@CajetanRodrigues Why have you hardcoded this? if you are fetching client details then put his/her name.

CajetanRodrigues replied:
Ok I will do that

CajetanRodrigues replied:
Done

ankit01ojha commented on src/app/services/authentication.service.ts:
```diff
@@ -1,36 +1,48 @@
-    // }
+   login(username: string, password: string): Observable<boolean> {
+      this.http.post<any[]>
+      ('https://mobile.openmf.org/fineract-provider/api/v1/self/authentication?username=' + `${username}` + '&password=' + `${password}`,
```
@CajetanRodrigues check.

CajetanRodrigues replied:
What's to be checked? Its the same api specified in the documentation

CajetanRodrigues replied:
> @CajetanRodrigues Login using a recently registered user is not working.

I works fine, you can check the gif for the demonstration. When the user signs up, he gets an email with his username and password, which he can then use to login. I have clearly demonstrated that in the gif above. Also, if the user registration is successful, a snackbar pops up at the bottom

CajetanRodrigues replied:
Working for all test cases now

ankit01ojha commented on src/app/services/user.service.ts:
```diff
@@ -1,10 +1,13 @@
   headers: new HttpHeaders({ 'Content-Type': 'application/json',
 'Fineract-Platform-TenantId': 'mobile',
-Authorization: 'Basic bWlmb3M6cGFzc3dvcmQ=' })
+Authorization: 'Basic bWlmb3M6cGFzc3dvcmQ=',
```
@CajetanRodrigues hardcoded.

CajetanRodrigues replied:
Ok I will resolve that

CajetanRodrigues replied:
> @CajetanRodrigues Login using a recently registered user is not working.

I have resolved that. You can now sign up and login the same credentials you put in the form. Although the authentication flow is temporary, we could fix it in the near future. For now this PR needs to get merged to have the base fixed once and for all. So that I as well as upcoming contributors can create new components on top of the base.

CajetanRodrigues commented:
> @CajetanRodrigues why `username: mifos and password: password` is working here, it should not work because there is no self service user as` mifos`. [openMF/web-self-service-app#27](https://github.com/openMF/web-self-service-app/pull/27)

Done

CajetanRodrigues commented:
I have resolved all the changes suggested.

CajetanRodrigues commented:
You can now sign up and login using the same credentials. Also I have fixed all the UI glitches that you have suggested.

CajetanRodrigues commented:
> @CajetanRodrigues login still has issues. When I login with right credentials it is first showing an invalid credential message and then it shows login successfully. And if a user is not a self-service user it is showing invalid credentials message rather than `not a self-service user`. Name of the client fetched is wrong. And could you change the UI of the landing page similar to the old one?

Re:
I have resolved all the login errors and stated proper login messages now. Also, the UI of the landing page is the same as the old app. Since this is Angular Material with Angular 7, the UI would look a bit different, but the rest of the things have been resolved and all the above changes have been taken into consideration.

@gauravsaini03 Can you take a final look?

gauravsaini03-zz commented on src/app/authentication/verification/verification.component.html:
```diff
@@ -1,9 +1,9 @@
 <form  #loginForm="ngForm" >
   <img src="https://www.google-melange.com/archive/gci/2014/orgs/mifos/logo-200.png" alt="No Image found">
-    <mat-form-field>
+    <mat-form-field class="font-style ">
```
Please remove space in the class

CajetanRodrigues replied:
okay sure

CajetanRodrigues replied:
Hi Gaurav, as you have suggested. I have indented all the code.

ankit01ojha commented:
> Also, the UI of the landing page is the same as the old app. Since this is Angular Material with Angular 7, the UI would look a bit different.

So, you mean that the size of the font, position of font, spacing, and colors in button are different because you are using Angular Material with Angular 7 and not because of CSS?  Modify the CSS to make it similar to the UI of the landing page of web self-service app because UI landing page of web self-service app is quite good.

CajetanRodrigues commented:
> > Also, the UI of the landing page is the same as the old app. Since this is Angular Material with Angular 7, the UI would look a bit different.
> 
> So, you mean that the size of the font, position of font, spacing, and colors in button are different because you are using Angular Material with Angular 7 and not because of CSS? Modify the CSS to make it similar to the UI of the landing page of web self-service app because UI landing page of web self-service app is quite good.

No, I didn't mean that. I will do the needful. Thanks for reviewing

CajetanRodrigues commented:
> @CajetanRodrigues you need to fetch the details of the client who has logged in and display their details in dashboard. I don't see you fetching the login clients details.
> 
> And do the change the UI of the login page, it's still the same.


+ Made the look of login page as similar as possible to the Online Banking App 2.0
+ Fetched display name & display picture of client from API and integrated with the app

CajetanRodrigues commented:
> @CajetanRodrigues Why there are so many unwanted undefined requests are being sent in the background? When being logged in for the first time the loader keeps on loading( maybe because of these unwanted error requests). Fix them. And why are you saving the image of the client in the local storage? Remove that too.

+ Removed unwanted undefined get requests
+ Removed image from local storage
+ Resolved the problem, where spinner loads on the first login and data was not able to load.

CajetanRodrigues commented:
> @CajetanRodrigues remove inline CSS.

+ Proof checked entire html code for inline css and removed wherever present. 
+ Added Stylings in external scss file.

CajetanRodrigues commented:
> @CajetanRodrigues Client's name is missing and charts are not correct for the client, tally it with the old app and fix it.

+ Fixed the issue that the client name was not visible after reloading of page, but available after immediate login. 
+ Tallied the data with the old app.

ankit01ojha commented on src/app/services/user.service.ts:
```diff
@@ -1,10 +1,13 @@
+  headers: new HttpHeaders({
+    'Content-Type': 'application/json',
+    'Fineract-Platform-TenantId': 'mobile',
+    Authorization: 'Basic bWlmb3M6cGFzc3dvcmQ='
```
@CajetanRodrigues hardcoded.

CajetanRodrigues replied:
+ I have done the changes. It now fetches the token dynamically.

CajetanRodrigues commented:
> @CajetanRodrigues I cannot see clients APIs being called. You need to fetch clients details because if it shows no content then only you can differentiate between a self service user and a non self service user.

+ You can find the client.service.ts file where I have defined methods for fetching client details
+ I have called the same methods in  app.component.ts , dashboard.component.ts & individual chart components

ankit01ojha commented on src/app/services/user.service.ts:
```diff
@@ -13,17 +16,33 @@ export class UserService {
+          firstname: user.firstname,
+          lastname: user.lastname,
+          email: user.email,
+          officeId: 1,
```
@CajetanRodrigues hardcoded, this is what I meant by the previous comment. Without calling client API how you are getting the id of a client for further calling of APIs and for checking whether a user is a client or not.
